BZ1850510 - Update NFS registry storage docs

bobfuru opened the pull request:
[BZ1850510](https://bugzilla.redhat.com/show_bug.cgi?id=1850510) - Add link to Group IDs module for recommended info about using supplemental groups for NFS shared storage. And also update NFS example to use `root_squash` and `no_wdelay`. Applies to OCP 4.3+.

Preview links - step 2 note and example now include `root_squash` and supplemental group note is added in step 1: 

- https://bz1850510--ocpdocs.netlify.app/openshift-enterprise/latest/registry/configuring_registry_storage/configuring-registry-storage-baremetal.html#registry-configuring-storage-baremetal_configuring-registry-storage-baremetal
- https://bz1850510--ocpdocs.netlify.app/openshift-enterprise/latest/registry/configuring_registry_storage/configuring-registry-storage-vsphere.html#installation-registry-storage-config_configuring-registry-storage-vsphere

bobfuru commented:
@huffmanca - PTAL and let me know if this approach makes sense to you. Still not able to add xrefs within the module itself, so the best solution seemed to add the `supplementalGroups` info at the end of the bare metal reg config procedure as an important note. I've also added `root_squash` in step 2 for NFS. Thanks!

huffmanca commented on registry/configuring_registry_storage/configuring-registry-storage-baremetal.adoc:
```diff
@@ -12,6 +12,11 @@ include::modules/installation-registry-storage-config.adoc[leveloffset=+1]
 
 include::modules/registry-configuring-storage-baremetal.adoc[leveloffset=+1]
 
+[IMPORTANT]
```
Would it be possible to include this in the module instead of at the assembly level? That way we could also get this included where the user is defining these options instead of at the end of the procedure.

bobfuru replied:
I agree it makes more sense in the module. I've added to step 1 (didn't want to add another note in step 2) and also included the xref to the supporting docs in the assembly so a user has link. Not ideal but should be better than having all the info after the procedure.

huffmanca replied:
This looks good to me!

bobfuru commented:
Thanks, @huffmanca! Changes made, could you please take a second look?

> In addition, while this addresses the issues in the baremetal assembly, we need to make identical changes to the vsphere assembly as well.

I've also added the note to the vSphere assembly, but I left out the corresponding NFS -> Group IDs link. We don't seem to have similar supplemental group content elsewhere for vSphere. Do you think this approach is sound?

huffmanca commented:
Thanks for these changes!

/lgtm

bobfuru commented:
@qinpingli - PTAL for QE review.

bobfuru commented:
@openshift/team-documentation PTAL

ahardin-rh commented:
LGTM :+1:

qinpingli commented on modules/registry-configuring-storage-vsphere.adoc:
```diff
@@ -34,6 +34,11 @@ a different storage backend, such as `NFS`, to configure the registry storage.
 +
+[NOTE]
+====
+When using shared storage such as NFS, it is strongly recommended to create Security Context Constraints (SCCs) that use `supplementalGroups` IDs, rather than `fsGroup` IDs. Refer to the NFS *Group IDs* documentation for details.
```
SCC has specific meaning in openshift, it's the api `securitycontextconstraints.security.openshift.io`, so here I think we should use `Security Context.`

`fsGroup` only can define one group ID.

qinpingli replied:
How about add a link to `Group IDs`?

bobfuru replied:
> SCC has specific meaning in openshift, it's the api `securitycontextconstraints.security.openshift.io`, so here I think we should use `Security Context.`
> 
> `fsGroup` only can define one group ID.

Hi, @qinpingli. I am glad you pointed that out. I changed the text a bit to be more in line with how we describe the strategy in the annotation note (8) used in the first example in this topic: https://docs.openshift.com/container-platform/4.5/authentication/managing-security-context-constraints.html#security-context-constraints-example_configuring-internal-oauth

bobfuru replied:
> How about add a link to `Group IDs`?

Would be nice, yes, but unfortunately we are unable to include xrefs at the "module" level as it can break docs builds. As a workaround, I included the link at the end (which is at the "assembly" level).

qinpingli commented:
@bobfuru  Thanks for the update.

Now the patch lgtm.

bobfuru commented on modules/registry-configuring-storage-vsphere.adoc:
```diff
@@ -42,13 +47,12 @@ $ oc get pod -n openshift-image-registry
-If the storage type is `NFS`, and you want to scale up the registry Pod by setting
-`replica>1` you must enable the `no_wdelay` mount option. For example:
-
+* If the storage type is `emptyDIR`, the replica number cannot be greater than `1`.
+* If the storage type is `NFS`, you must enable the `no_wdelay` and `root_squash` mount options. For example:
++
```
@huffmanca and @qinpingli - I made a small change here (removing ref in NFS for `replica>1`) based on Alex's BZ comment: https://bugzilla.redhat.com/show_bug.cgi?id=1850510#c15

PTAL and let me know if you both are good with this revision. Thanks.

qinpingli replied:
LGTM

huffmanca replied:
This looks good to me!

bobfuru commented:
/cherrypick enterprise-4.6

bobfuru commented:
/cherrypick enterprise-4.5

bobfuru commented:
/cherrypick enterprise-4.4

bobfuru commented:
/cherrypick enterprise-4.3
